uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/GroupRESTController.java

```diff
@@ -45,8 +47,12 @@ public void setPersonManager(IPersonManager personManager) {
 
     @RequestMapping(value = "/groups", method = RequestMethod.GET)
     public ModelAndView getUsersGroup(HttpServletRequest request, HttpServletResponse response) {
-        Set<Entity> groups =
-                groupService.getGroupsForMember(personManager.getPerson(request).getUserName());
+        IPerson person = personManager.getPerson(request);
+        Set<Entity> groups = Collections.EMPTY_SET;
+        if (person != null) {
+            groups = groupService.getGroupsForMember(person.getUserName());
+        }
+
```

Comment:
How is guest user handled?

Might also want to check that the return value from getUserName() is not blank.